Functions_and_Scope_Lab.js

```diff
@@ -0,0 +1,370 @@
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+//
+//
+// ## Question 9
+console.log("_____#9_____")
+//
+let forwardArr = [1, 2, 3, 4]
+
+const reverse = (arr) => {
+  let reverseArr = []
+  for (let i = arr.length - 1; i >= 0; i--) {
+  reverseArr.push(arr[i])
+ }
+ return reverseArr
+}
+
+let return9 = reverse(forwardArr)
+console.log(return9)
+
+// Write a function named `reverse` that takes an array as a parameter. The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+//
+//
+// ## Question 10
+console.log("_____#10_____")
+//
+// Write a function that prints out the most frequently occurring element in an array.
+//
+
+
+// Example:
+//
+let input2 = [2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]
+const freqElementArr = (arr) => {
+  let counter = {}
+  let max = arr[0]
+  let maxCount = 1
+  for (let i = 0; i < arr.length; i++) {
+    let value = arr[i]
+    if (counter[value] == null) {
+      counter[value] = 1
+    } else {
+      counter[value]++
+    }
+    if (counter[value] > maxCount){
+      max = value
+      maxCount = counter[value]
+    }
+  }
+  return max
+}
+
+let return10 = freqElementArr(input2)
+console.log(return10)
+
+// NOTE: I'm INCREDIBLY lost on this one. The code doesn't brake, but it doesn't
+// return anything either.
+
+//
+// Output: `2`
+//
+//
+//
+// ## Question 11
+console.log("_____#11_____")
+//
+// Write a function that sums all the even numbers in an array.
+//
+// Example:
+//
+let input1 = [1,2,5,64,23,22,91,3]
+
+const sumOfEven = (arr) => {
+  let sum2 = 0
+  for(let i = 0; i < arr.length; i++) {
+    if (arr[i] % 2 === 0) {
+    sum2 += arr[i]
+    console.log(sum2);
+    }
+  }
+  return sum2
+}
+let return11 = sumOfEven(input1)
+console.log(return3)
+
+
+//
+// Output: `88`
+//
+//
+// ## Question 12
+console.log("_____#12_____")
+//
+// Write a function that flips an object.  All of the keys are now values and all of the values are now keys.
+//
+
+let spanishCountingObj = {
+  1: "Uno",
+  5: "Cinco"
+}
+const objKeyValSwap = (obj) => {
+  let swapCopy = {}
+  let copyValues = []
+  let copyKeys = []
+  for (let values in obj) {
+    copyValues.push(obj[values])
+  }
+  for (let keys in obj) {
+    copyKeys.push(keys)
+  }
+  for (let i = 0; i < copyValues.length; i++) {
+     if (copyValues !== i) {
+       swapCopy = copyValues[i]
+     }
+  }
+  //for (let i = 0; i )
+ return swapCopy
+}
+
+console.log(objKeyValSwap(spanishCountingObj))
+
+// Example:
+//
+// Input: `{1: "uno", 5: "cinco"}`
+//
+// Output: `{"uno": 1, "cinco": 5}`
+//
+//
+// ## Question 13
+console.log("_____#13_____")
+//
+// Write a function that determines if a value is inside of array.
+//
+let exeArr = [2,1,4,7,6,9]
+const arrValCheck = (arr, num) => {
```

Comment:
When you run this function it would not work as it only returns true or false after one iteration of the loop. That is because having return statement inside a loop is a very bad idea. Avoid putting return inside a loop when possible.